tests/stream_test.cc

```diff
@@ -2,27 +2,51 @@
 
 #include "gtest/gtest.h"
 
+#include <climits>
+#include <cstdio>
+#include <cstdlib>
 #include <cstring>
+#include <fstream>
+#include <iostream>
+#include <string>
 
 using namespace Pistache;
 
 TEST(stream, test_buffer)
 {
     char str[] = "test_string";
     size_t len = strlen(str);
-    Buffer buffer1(str, len, false);
+    RawBuffer buffer1(str, len, false);
 
-    Buffer buffer2 = buffer1.detach(0);
-    ASSERT_EQ(buffer2.length, len);
-    ASSERT_EQ(buffer2.isDetached, true);
+    RawBuffer buffer2 = buffer1.detach(0);
+    ASSERT_EQ(buffer2.size(), len);
+    ASSERT_EQ(buffer2.isDetached(), true);
 
-    Buffer buffer3;
-    ASSERT_EQ(buffer3.length, 0u);
-    ASSERT_EQ(buffer3.isDetached, false);
+    RawBuffer buffer3;
+    ASSERT_EQ(buffer3.size(), 0u);
+    ASSERT_EQ(buffer3.isDetached(), false);
 
-    Buffer buffer4 = buffer3.detach(0);
-    ASSERT_EQ(buffer4.length, 0u);
-    ASSERT_EQ(buffer4.isDetached, false);
+    RawBuffer buffer4 = buffer3.detach(0);
+    ASSERT_EQ(buffer4.size(), 0u);
+    ASSERT_EQ(buffer4.isDetached(), false);
     
     ASSERT_THROW(buffer1.detach(2 * len);, std::range_error);
+}
+
+TEST(stream, test_file_buffer)
+{
+    char fileName[PATH_MAX] = "/tmp/tmpfileXXXXXX";
+    mkstemp(fileName);
+    std::cout << "Temporary file name: " << fileName << '\n';
+
+    const std::string dataToWrite("Hello World!");
+    std::ofstream tmpFile;
+    tmpFile.open(fileName);
+    tmpFile << dataToWrite;
+    tmpFile.close();
+
```

Comment:
Please unlink the temp file when you are done, and consider putting "pistacheio" in the filename.

Reply:
@dennisjenkins75 thank you so much for review, I've done it.